frontend/src/router/middleware/canVisit.ts

```diff
@@ -0,0 +1,159 @@
+import { computed, inject, type Ref } from "vue";
+import type { Middleware, MiddlewareContext } from "./index";
+import { QueryClient } from "@tanstack/vue-query";
+import useIsAdmin from "@/composables/useIsAdmin";
+import useIsTeacher from "@/composables/useIsTeacher";
+import { useSubjectsQuery } from "@/queries/Subject";
+import { useProjectQuery, useProjectsQuery } from "@/queries/Project";
```

Comment:
```suggestion
import { useProjectsQuery } from "@/queries/Project";
```

'useProjectQuery' is defined but never used